tests/runner/test_runner.py

```diff
@@ -1062,3 +1085,88 @@ async def test_deploy_with_image_string_no_tag(
         )
         assert used_name == "test-registry/test-image"
         assert used_tag is not None
+
+    async def test_deploy_with_partial_success(
+        self,
+        mock_build_image,
+        mock_docker_client,
+        mock_generate_default_dockerfile,
+        work_pool_with_image_variable,
+        capsys,
+    ):
+        deployment_ids = await deploy(
+            await dummy_flow_1.to_deployment(
+                __file__, job_variables={"image_pull_policy": "blork"}
+            ),
+            await (
+                await flow.from_source(
+                    source=MockStorage(), entrypoint="flows.py:test_flow"
+                )
+            ).to_deployment(__file__),
+            work_pool_name=work_pool_with_image_variable.name,
+            image="test-registry/test-image",
+        )
+        assert len(deployment_ids) == 1
+
+        console_output = capsys.readouterr().out
+        assert (
+            "Encountered errors while creating/updating deployments" in console_output
+        )
+        assert "failed" in console_output
+        # just the start of the error due to wrapping
+        assert "'blork' is not one" in console_output
+
+    async def test_deploy_with_complete_failure(
+        self,
+        mock_build_image,
+        mock_docker_client,
+        mock_generate_default_dockerfile,
+        work_pool_with_image_variable,
+        capsys,
+    ):
+        deployment_ids = await deploy(
+            await dummy_flow_1.to_deployment(
+                __file__, job_variables={"image_pull_policy": "blork"}
+            ),
+            await (
+                await flow.from_source(
+                    source=MockStorage(), entrypoint="flows.py:test_flow"
+                )
+            ).to_deployment(__file__, job_variables={"image_pull_policy": "blork"}),
+            work_pool_name=work_pool_with_image_variable.name,
+            image="test-registry/test-image",
+        )
+        assert len(deployment_ids) == 0
+
+        console_output = capsys.readouterr().out
+        assert (
+            "Encountered errors while creating/updating deployments" in console_output
+        )
+        assert "failed" in console_output
+        # just the start of the error due to wrapping
+        assert "'blork' is not one" in console_output
+
+        assert "prefect worker start" not in console_output
```

Comment:
Should we also assert that "To execute flow runs from these deployments" is not in the output?